crates/decoder/src/decoder.rs

```diff
@@ -271,25 +275,28 @@ impl From<Option<u64>> for EndBlock {
 pub fn stream_blocks(
     reader: Reader,
     end_block: EndBlock,
-) -> Result<impl Iterator<Item = Block>, DecoderError> {
+) -> Result<impl Iterator<Item = AnyBlock>, DecoderError> {
     let mut current_block_number = 0;
 
     let mut reader = reader.into_reader()?;
     let end_block = end_block.block_number();
 
     let mut blocks = Vec::new();
 
+    let header = DbinHeader::try_from_read(&mut reader)?;
+    let content_type: ContentType = header.content_type().as_str().try_into()?;
```

Comment:
See once you change the api to return a reference you won't need to call `as_str` here.